File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOTemperatureSwitchParamModel.kt
//
//  IDOTemperatureSwitchParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.annotations.SerializedName

import com.google.gson.Gson
import com.google.gson.GsonBuilder

///
/// Set Night-time Temperature Switch Event Code

open class IDOTemperatureSwitchParamModel(
    mode: Int,
    startHour: Int,
    startMinute: Int,
    endHour: Int,
    endMinute: Int,
    unit: Int
) : IDOBaseModel {

    /// Mode:
    /// 1: On
    /// 0: Off
    @SerializedName("mode")
    var mode: Int = mode

    /// Start time, hour
    @SerializedName("start_hour")
    var startHour: Int = startHour

    /// Start time, minute
    @SerializedName("start_minute")
    var startMinute: Int = startMinute

    /// End time, hour
    @SerializedName("end_hour")
    var endHour: Int = endHour

    /// End time, minute
    @SerializedName("end_minute")
    var endMinute: Int = endMinute

    /// Temperature unit setting:
    /// 1 :Celsius
    /// 2 :Fahrenheit
    @SerializedName("unit")
    var unit: Int = unit


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOTemperatureSwitchParamModel.swift
//
//  IDOTemperatureSwitchParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set Night-time Temperature Switch Event Code
@objcMembers
public class IDOTemperatureSwitchParamModel: NSObject, IDOBaseModel {
    /// Mode:
    /// 1: On
    /// 0: Off
    public var mode: Int
    /// Start time, hour
    public var startHour: Int
    /// Start time, minute
    public var startMinute: Int
    /// End time, hour
    public var endHour: Int
    /// End time, minute
    public var endMinute: Int
    /// Temperature unit setting:
    /// 1 :Celsius
    /// 2 :Fahrenheit
    public var unit: Int
    
    enum CodingKeys: String, CodingKey {
        case mode = "mode"
        case startHour = "start_hour"
        case startMinute = "start_minute"
        case endHour = "end_hour"
        case endMinute = "end_minute"
        case unit = "unit"
    }
    
    public init(mode: Int,startHour: Int,startMinute: Int,endHour: Int,endMinute: Int,unit: Int) {
        self.mode = mode
        self.startHour = startHour
        self.startMinute = startMinute
        self.endHour = endHour
        self.endMinute = endMinute
        self.unit = unit
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

